libr/bin/bfile.c

```diff
@@ -875,37 +874,42 @@ R_API int r_bin_file_hash(RBin *bin, ut64 limit, const char *file) {
 	buf_len = r_io_desc_size (iod);
 	// By SLURP_LIMIT normally cannot compute ...
 	if (buf_len > limit) {
-		eprintf ("Cannot compute hash\n");
-		return -1;
-	}
-	//  XXX should use io api not raw file slurping
-	int blen = 0;
-	buf = (ut8*)r_file_slurp (file, &blen);
-	if (!buf) {
+		eprintf ("Cannot compute hash: %zu/%zu\n", buf_len, limit);
 		return false;
 	}
-	buf_len = blen;
-	if (!o->info) {
+#define BLK_SIZE_OFF 64000
+	ut8 *buf = malloc (BLK_SIZE_OFF);
+	if (!buf) {
+		eprintf ("Cannot allocate computation buffer\n");
 		return false;
 	}
-	if (buf) {
-		ctx = r_hash_new (false, R_HASH_MD5 | R_HASH_SHA1);
-#define BLK_SIZE_OFF 1024
-		for (i = 0; i < buf_len; i += BLK_SIZE_OFF) {
-			(void)r_hash_do_md5 (ctx, &buf[i], R_MIN (buf_len-i, BLK_SIZE_OFF));
-			(void)r_hash_do_sha1 (ctx, &buf[i], R_MIN (buf_len-i, BLK_SIZE_OFF));
-		}
-		r_hash_do_end (ctx, R_HASH_MD5);
-		p = hash;
-		r_hex_bin2str (ctx->digest, R_HASH_SIZE_MD5, p);
-		o->info->hashes = r_strbuf_new ("");
-		r_strbuf_appendf (o->info->hashes, "md5 %s", hash);
-		r_hash_do_end (ctx, R_HASH_SHA1);
-		p = hash;
-		r_hex_bin2str (ctx->digest, R_HASH_SIZE_SHA1, p);
-		r_strbuf_appendf (o->info->hashes, "\nsha1 %s", hash);
-		r_hash_free (ctx);
-	}
+	ctx = r_hash_new (false, R_HASH_MD5 | R_HASH_SHA1);
+	while (r+BLK_SIZE_OFF < buf_len) {
+		r_io_desc_seek (iod, r, R_IO_SEEK_SET);
+		size_t rem_len = R_MIN (buf_len-r, BLK_SIZE_OFF);
```

Comment:
there's no remaining necessary at all, inside this loop the blocks are always of BLK_SIZE_OFF (even if thats a bad name)

Reply:
this is const